src/test/requestUser.test.js

```diff
@@ -0,0 +1,263 @@
+  });
+
+  it("test 2: Add user to accepted list both users.", async () => {
+    const toUserID = "12345";
+    const fromUserID = "ABCDE";
+
+    const querySnapshot = {
+      forEach: (fn) => {
+        fn({
+          data: () => [
+            {
+              user_id: "12345",
+              pending_list: ["Walter", "Joe"],
+              accepted_list: ["White"],
+            },
+          ],
+        });
+      },
+      empty: (fn) => true,
+    };
+    getDocs.mockImplementation(() => querySnapshot);
+
+    // Call the tested function.
+    await acceptRequest(toUserID, fromUserID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", toUserID)
+    );
+    expect(addDoc).toBeCalledTimes(2);
+    expect(collection).toHaveBeenCalledWith(tableName);
+  });
+
+  it("Get list of Accepted Request", async () => {
+    const userID = "12345";
+    const acceptedList = [
+      new Map([["accepted_list", ["12345", "12346", "12356"]]]),
+    ];
+    const userIDList = ["12345", "12346", "12356"];
+    getDocs.mockImplementation(() => acceptedList);
+
+    // Call the tested function.
+    let testReturn = await userAcceptedList(userID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", userID)
+    );
+    expect(collection).toHaveBeenCalledWith(tableName);
+    expect(testReturn).toEqual(userIDList);
+  });
+
+  it("Get list of Pending Request", async () => {
+    const userID = "12345";
+    const pendingList = [
+      new Map([["pending_list", ["12345", "12346", "12356"]]]),
+    ];
+    const userIDList = ["12345", "12346", "12356"];
+    getDocs.mockImplementation(() => pendingList);
+
+    // Call the tested function.
+    let testReturn = await userPendingList(userID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", userID)
+    );
+    expect(collection).toHaveBeenCalledWith(tableName);
+    expect(testReturn).toEqual(userIDList);
+  });
+
+  it("Check whether the user request is accepted.", async () => {
+    const userID = "12345";
+    const fromUserID = "12356";
+    const acceptedList = [
+      new Map([["accepted_list", ["12345", "12346", "12356"]]]),
+    ];
+
+    getDocs.mockImplementation(() => acceptedList);
+
+    // Call the tested function.
+    let testReturn = await checkUserRequest(userID, fromUserID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", userID)
+    );
+    expect(collection).toHaveBeenCalledWith(tableName);
+    expect(testReturn).toEqual(true);
```

Comment:
can we add a test case here where we query with an ID that's *not* in the list to make sure it returns false?

Reply:
This will be a separate test case but it does not increase our coverage.